src/graph.rs

```diff
@@ -501,32 +501,24 @@ where
     /// It is currently used if filtering of nodes is required.
     pub fn get_filtered_adjacency_indices(&self, nodes_to_keep: &Vec<T>) -> Vec<Vec<usize>> {
         let num_nodes = nodes_to_keep.len();
-        let mut indices = Vec::new();
-        for _ in 0..num_nodes {
-            indices.push(Vec::new());
+        let mut indices = Vec::with_capacity(num_nodes);
+        let mut node_map = HashMap::with_capacity(num_nodes);
+        for n in 0..num_nodes {
+            // make initial capacity 10% of total
+            indices.push(Vec::with_capacity(num_nodes / 10));
```

Comment:
Just wonder if there shouldn't be called `shrink_to_fit` at some point (after adding all indices) to be sure we're not wasting memory as we'd like to load big files with network descriptions (like 20MB) and make some copies of them.

Reply:
`shrink_to_fit`, TIL
like before returning the `Vec<Vec<usize>>`, we shrink the capacity of both inner and outer Vector?

Reply:
sure, sounds good to me.  But we only need it on the inner vectors, i.e., the ones with a single node's connection.  The outer vector length will always be the same same (num_nodes).

Reply:
`TIL`, TIL  😄

Reply:
done